src/app/Main/Body/QueryPage/QueryPage-Container.component.tsx

```diff
@@ -251,106 +244,18 @@ export function QueryPageContainerComponent(): JSX.Element {
         setFilter(newFilter);
     };
 
+    // handle export
+
     const fetchIsolatesAndExportZip = async (): Promise<void> => {
```

Comment:
See comment in the old branch: E.g. this is a mixed responsibility and could be separated in fetching and exporting (simplified):
```
const fetchIsolatesAndExportZip = () => {
  setIsBusy(true);
  checkConditions...
  fetchIsolateParams = from url params and state
  isolates = fetchIsolates(fetchIsolateParams);
  zipCsv = mapIsolatesToCsvString(isolates)
  exportZip(zipCsv)
  setIsBusy(false);
}
```
And the whole thing could be live in an export container.

Reply:
As there is an extra ticket for refactoring the containers. I would note that for later.